src/platform_impl/linux/wayland/header.rs

```diff
@@ -0,0 +1,36 @@
+use gtk::{prelude::*, ApplicationWindow, EventBox, HeaderBar};
+
+pub struct WlHeader;
+
+impl WlHeader {
+  pub fn setup(window: &ApplicationWindow, title: &str) {
+    let header = HeaderBar::builder()
+      .show_close_button(true)
+      .decoration_layout("menu:minimize,maximize,close")
+      .title(title)
+      .build();
+
+    let event_box = EventBox::new();
```

Comment:
how about instead of using an `EventBox` we use the `HeaderBar` but call `headerbar.add_events(EventMask::BUTTON_PRESS_MASK)`

- https://docs.rs/gtk/latest/gtk/prelude/trait.WidgetExtManual.html#method.add_events
- https://docs.rs/gdk/0.18.0/gdk/struct.EventMask.html

Reply:
I’ve tried this before, but to show you directly, it doesn’t work.
![event-mask](https://github.com/user-attachments/assets/ac2ed1f9-de45-47c4-b92a-49045f1da533)

Reply:
ah, nvm mind then